daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -192,8 +219,50 @@ func (l *logStream) Close() error {
 	return nil
 }
 
-// create creates a log stream for the instance of the awslogs logging driver
+// create creates log group and log stream for the instance of the awslogs logging driver
 func (l *logStream) create() error {
+	if err := l.createLogStream(); err != nil {
+		if l.logCreateGroup {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == resourceNotFoundCode {
```

Comment:
> Ah, right, so checking if a group exists actually requires different permissions than using a group.

Yes.  You can see the IAM permissions available for CloudWatch Logs in the [developer guide](http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/iam-identity-based-access-control-cwl.html).  In general, AWS services tend to allow policies that specify individual APIs to access, so that different scenarios can be supported.  One relevant example that fits both logs and metrics is write-only, so that a host which is producing metrics or producing logs cannot read metrics or logs from other resources in the same account.